beacon-chain/rpc/proposer_server.go

```diff
@@ -65,11 +70,38 @@ func (ps *ProposerServer) RequestBlock(ctx context.Context, req *pb.BlockRequest
 		return nil, errors.Wrap(err, "could not get parent block signing root")
 	}
 
-	// Construct block body
-	// Pack ETH1 deposits which have not been included in the beacon chain
-	eth1Data, err := ps.eth1Data(ctx, req.Slot)
-	if err != nil {
-		return nil, errors.Wrap(err, "could not get ETH1 data")
+	var eth1Data *ethpb.Eth1Data
+	if ps.mockEth1Votes {
```

Comment:
Review this

Reply:
Can you add a test for this whole logic ? this path isnt tested